### DISCUSSION TITLE
Environment Secrets created through Terraform are ... empty?

### DISCUSSION THREAD

**Discussion by bobthemighty** [community/community#29706] [Actions] [answered]
Hi, I have a weird problem.

I've set up a Github Action to configure Github Environments. I use the Github Terraform provider to create the environments and add secrets to them.

The secrets are created, in that I can go and look at them in the UI, but when I try to access them in a job, they seem to be empty.
If I have the same secret defined at the repository level and in an environment, the repository secret is applied. If I `echo ${{secrets.whatever}} || base64` I get an empty string.

If I set the secret value manually, everything works correctly. From what I can see, if I set TF_LOG=DEBUG, the secret is being created, and is passing the correct `encrypted_value` to the API. I'm stumped.

**Comment by chrispat** [comment, marked as answer]
I think this issue may be more appropriate for the github provider terraform repository.  I know the gh cli command for setting an environment secret properly https://cli.github.com/manual/gh_secret_set and that just uses the public secrets api.

**Reply by bobthemighty** [reply]
I've raised an issue there, too, but the level of activity on the repo doesn't fill me with confidence. Moreover, from what I can see, the provider _is_ creating a secret and passing the value correctly. Maybe I should try with the gh client.

Edit: `gh` is setting repository secrets, not environment secrets so I can't try that.

**Reply by chrispat** [reply]
I just tested this on my repository and it seems to be working for my repository.  I did have to make sure the environment existed prior to creating the secret or it threw an error.
![image](https://user-images.githubusercontent.com/185122/184953876-d3ef0f9b-b4c8-49cc-9ecd-2f8b4cfb0355.png)
![image](https://user-images.githubusercontent.com/185122/184953962-6b6db6c1-d617-435d-9977-694e024827b2.png)

**Reply by bobthemighty** [reply]
Beautiful thanks @chrispat - I used that and can confirm that the `gh` tool works perfectly. Secrets created with that tool _are_ immediately accessible through actions. I'll take this over to the terraform provider repo :)

**Comment by bobthemighty** [comment]
@chrispat you actually helped me solve this one. I wasn't using the `--env` flag when generating the secrets with `gh` for use in Terraform. As a result, the secrets were invalid.
